bgpd/bgpd.c

```diff
@@ -1247,12 +1248,12 @@ struct peer *peer_new(struct bgp *bgp)
 	 * bounds checking for every single attribute as we construct an
 	 * UPDATE.
 	 */
-	peer->obuf_work =
-		stream_new(BGP_MAX_PACKET_SIZE + BGP_MAX_PACKET_SIZE_OVERFLOW);
-	peer->ibuf_work =
-		ringbuf_new(BGP_MAX_PACKET_SIZE * BGP_READ_PACKET_MAX);
+	peer->obuf_work = stream_new(BGP_MAX_EXT_MESSAGE_PACKET_SIZE
+				     + BGP_MAX_PACKET_SIZE_OVERFLOW);
+	peer->ibuf_work = ringbuf_new(BGP_MAX_EXT_MESSAGE_PACKET_SIZE
+				      * BGP_READ_PACKET_MAX);
```

Comment:
Handling of this buffer should be done based on if feature is enabled or not. If we do not need this feature then we are holding up memory here which is not being used at all?

Reply:
Yes, I don't know how to proceed here. The main issue is that this is done on peer create. But the size of the message is negotiated during OPEN and this can't be reallocated. Do you have some suggestions on how to overcome it here?

Reply:
Weellll....we could turn off I/O, create a new ringbuf, copy the contents, swap the pointers, destroy the old one and then turn I/O back on. Doesn't make me real happy though.

Reply:
Gave a thought and it does look like adding more code to handle this extra memory that is being allocated but not used when this feature is disabled.

Reply:
Seeing as we have effectively no support for dynamic capabilities I believe you have to bounce the session anyway for it to take effect, which is done by sending a notify in the cli handler